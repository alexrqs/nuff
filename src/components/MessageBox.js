import React from 'react';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';

import { PRIORITY } from '../utils/enums';

const useStyles = makeStyles((theme) => ({
  button: {
    backgroundColor: 'transparent',
    border: 'none',
    cursor: 'pointer'
  },

  buttonContainer: {
    textAlign: 'right'
  },

  paper: {
    padding: 10,
    marginBottom: 10,
  }
}));

const bgColor = {
  1: '#F56236',
  2: '#FCE788',
  3: '#88FCA3',
}
const MessagesBox = ({ messages, onClear, priority }) => {
  const css = useStyles();

  return (
    <div>
      Count {messages.length}
      {messages.map(({ message }, idx) =>
        <Paper key={idx} style={{  backgroundColor: bgColor[priority],  }} className={css.paper}>
          <div>{message}</div>
          <div className={css.buttonContainer}>
            <button className={css.button} onClick={() => onClear({message})}>Clear</button>
          </div>
        </Paper>
      )}
    </div>
  )
}

export default React.memo(MessagesBox)
