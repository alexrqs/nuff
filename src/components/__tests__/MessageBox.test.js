import MessageBox from '../MessageBox'
import renderer from 'react-test-renderer'

it('renders one message', () => {
  const messages = [
    { message: 'hi there', priority: 1}
  ]

  const component = renderer.create(
    <MessageBox messages={messages} ></MessageBox>
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('renders multiple messages', () => {
  const messages = [
    { message: 'hi there', priority: 1 },
    { message: 'hi new message', priority: 1 }
  ]

  const component = renderer.create(
    <MessageBox messages={messages} ></MessageBox>
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
