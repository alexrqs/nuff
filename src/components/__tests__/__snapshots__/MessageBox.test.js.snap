// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders multiple messages 1`] = `
<div>
  Count 
  2
  <div
    className="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <div>
      hi there
    </div>
    <div
      className="makeStyles-buttonContainer-2"
    >
      <button
        className="makeStyles-button-1"
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </div>
  <div
    className="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <div>
      hi new message
    </div>
    <div
      className="makeStyles-buttonContainer-2"
    >
      <button
        className="makeStyles-button-1"
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </div>
</div>
`;

exports[`renders one message 1`] = `
<div>
  Count 
  1
  <div
    className="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <div>
      hi there
    </div>
    <div
      className="makeStyles-buttonContainer-2"
    >
      <button
        className="makeStyles-button-1"
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </div>
</div>
`;
